### ISSUE TITLE
xSPExcelServiceApp allows for only one run - subsequent runs fail

### ISSUE THREAD

**Issue by RonTatro** [PowerShell/xSharePoint#251]
We're invoking the DscResource from within a Chef run and the first run creates the Excel Services App just fine but subsequent runs end in failure stating that the object already exists under the parent named ""

Specific Error Message: 
```
Invoke-DscResource : An object of the type Microsoft.Office.Excel.Server.MossHost.ExcelServerWebServiceApplication 
       named "Insert Your Name Here" already exists under the parent 
       Microsoft.Office.Excel.Server.MossHost.ExcelServerWebService named "".  Rename your object or delete the existing 
       object.
```

**Comment by BrianFarnhill** [comment]
@RonTatro I'm not totally familiar with how chef works but we can have a crack at this. Does chef still generate a MOF file with the instructions here? If so can you share the section of the MOF file you are using so I can try to replicate it in one of my test labs?

**Comment by RonTatro** [comment]
Thanks @BrianFarnhill Chef "DSC_Resource" resource just calls the DSC Resource directly on the server.  If it's helpful I'm pasting the .Mof on the server below.  LMK if you have any ideas on what could be wrong.

```
/*
**Description**

This resource is responsible for creating Excel Services Application instances within the local SharePoint farm.
The resource will provision and configure the Excel Services Service Application.
**Example**
    xSPExcelServiceApp ExcelServices
    {
        Name = "Excel Services Service Application"
        ApplicationPool = "SharePoint Web Services"
        InstallAccount  = $InstallAccount
    }
*/
[ClassVersion("1.0.0.0"), FriendlyName("xSPExcelServiceApp")]
class MSFT_xSPExcelServiceApp : OMI_BaseResource
{
    [Key, Description("The name of the service application")] string Name;
    [Required, Description("The name of the application pool to run the service app in")] string ApplicationPool;
    [Write, Description("Present ensures service app exists, absent ensures it is removed"), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] string Ensure;
    [Write, Description("POWERSHELL 4 ONLY: The account to run this resource as, use PsDscRunAsAccount if using PowerShell 5"), EmbeddedInstance("MSFT_Credential")] String InstallAccount;
};
```

**Comment by BrianFarnhill** [comment]
@RonTatro thats the schema MOF from the resource. Normally there will be a MOF file that has the actual instructions that are handed to the DSC engine - so I'm thinking maybe chef does that under the hood or something. Leave it with me either way, I'll try to repro it and let you know how I go.

**Comment by RonTatro** [comment]
@BrianFarnhill Looks like the TypeName may be incorrect for the Excel Service App:

This is what's in the code now: `{ $_.TypeName -eq "Excel Services Application" }`
In our environment, the TypeName appears to be:  `{$_.TypeName -eq "Excel Services Application Web Service Application" }`

This is resulting in the first run to succeed but then not find a match on subsequent runs and try to create the same Excel Service App again...resulting in failure.

**Comment by BrianFarnhill** [comment]
Confirmed - will get the fix in. Thanks @RonTatro

**Comment by BrianFarnhill** [comment]
The fix is in #265 - I'll close this after it's reviewed and merged to dev.

**Comment by BrianFarnhill** [comment]
This just got merged in to dev - closing this issue. Thanks for reporting it!
